package/src/package/serialization/mod.rs

```diff
@@ -111,7 +114,21 @@ impl Deserializable for Package {
         // Read manifest
         let manifest = PackageManifest::read_from(source)?;
 
-        Ok(Self { name, mast, manifest })
+        // Read optional account component metadata
+        let account_component_metadata_bytes: Option<Vec<u8>> = if source.has_more_bytes() {
+            Deserializable::read_from(source)?
+        } else {
+            // For compatibility with the packages serialized with the previous version without
+            // the AccountComponentMetadata bytes
+            None
+        };
```

Comment:
nit: we don't need to add complexity to support old versions, as we're still defining what the first version is

Reply:
I added it to keep `cli_*_masp` [tests](https://github.com/0xPolygonMiden/miden-vm/blob/greenhat/package-add-account-component-metadata/miden/tests/integration/cli/cli_test.rs#L45-L79) green, since they're using the compiled Miden package.

Reply:
I think we should recompile the Miden package in the the test, no? Basically until v1.0 of the package, we'd need to do this every time we break the package format.

Reply:
Done. In general, I agree, we should recompile. I thought that after #1721 we will rewrite `Package` regardless of what decision we make. 

Recompiling arbitrary programs from other tests without their sources is not a trivial task. In this case, I recognized it by name (`is_prime`), located the sources and recompiled it. But it might not be feasible in the future. We at least need to require the link to the sources when accepting compiled packages into the test suite.

While updating the `is_prime` package, I found and fixed the bug in the `run` command in https://github.com/0xPolygonMiden/miden-vm/pull/1731/commits/c60691bc8cd5ea34ba8f80ca32ccca89208af5dc (missing `stdlib`) and made https://github.com/0xPolygonMiden/miden-vm/issues/1754 to check the package's `stdlib` dependency.